app/jobs/get_an_identity_id_sync_job.rb

```diff
@@ -0,0 +1,67 @@
+class GetAnIdentityIdSyncJob < ApplicationJob
+  # Set as low priority so that these jobs don't block other time sensitive issue
+  queue_as :low_priority
+
+  def perform(user:)
+    if user.get_an_identity_id.blank?
+      create_skipped_ecf_sync_request_log(user:)
+      return false
+    end
+
+    ecf_user = user.ecf_user
+
+    if ecf_user.get_an_identity_id == user.get_an_identity_id
+      create_successful_ecf_sync_request_log(user:, message: "User's get_an_identity_id already synced to ECF")
+    else
+      # Sync get_an_identity_id to ECF
+      ecf_user.update!(get_an_identity_id: user.get_an_identity_id)
+
+      create_successful_ecf_sync_request_log(user:, message: "User's get_an_identity_id synced to ECF")
+    end
+
+    # Mark user as synced so that this user gets skipped in future sync runs
+    user.update_column(:get_an_identity_id_synced_to_ecf, true)
```

Comment:
I know we have in the rake task but do we want to add protection in this class to skip if a user is passed in with column true?

Reply:
I considered this, I decided to leave it without protections because in the case we want to manually kick off this job it would be a blocker to have it auto-skip if this was true. There's no *real* downside to letting it run, it's purely used in the rake task for speeding things up.